File: app/src/main/java/de/dreier/mytargets/utils/multiselector/HeaderBindingHolder.java
/*
 * Copyright (C) 2017 Florian Dreier
 *
 * This file is part of MyTargets.
 *
 * MyTargets is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License version 2
 * as published by the Free Software Foundation.
 *
 * MyTargets is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 */

package de.dreier.mytargets.utils.multiselector;

import android.view.View;

public abstract class HeaderBindingHolder<T> extends ItemBindingHolder<T> {

    /**
     * Constructor for header items
     *
     * @param itemView Header view
     */
    public HeaderBindingHolder(View itemView) {
        super(itemView);
    }

    @Override
    public void onClick(View v) {
    }

    /**
     * Returns whether {@link #itemView} is currently in a
     * selectable mode.
     *
     * @return True if selectable.
     */
    public boolean isSelectable() {
        return false;
    }

    /**
     * Does nothing.
     *
     * @param isSelectable True if selectable.
     */
    public void setSelectable(boolean isSelectable) {
    }

    @Override
    protected void onRebind() {
    }

    /**
     * Calls through to {@link #itemView#isActivated}.
     *
     * @return True if the view is activated.
     */
    public boolean isActivated() {
        return itemView.isActivated();
    }

    /**
     * Calls through to {@link #itemView#setActivated}.
     *
     * @param isActivated True to activate the view.
     */
    public void setActivated(boolean isActivated) {
        itemView.setActivated(isActivated);
    }
}

File: app/src/main/java/de/dreier/mytargets/utils/multiselector/HeaderBindingHolder.kt
/*
 * Copyright (C) 2018 Florian Dreier
 *
 * This file is part of MyTargets.
 *
 * MyTargets is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License version 2
 * as published by the Free Software Foundation.
 *
 * MyTargets is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 */

package de.dreier.mytargets.utils.multiselector

import android.view.View

/**
 * Constructor for header items
 *
 * @param itemView Header view
 */
abstract class HeaderBindingHolder<T>(itemView: View) : ItemBindingHolder<T>(itemView) {

    override fun onClick(v: View) {}

    @Suppress("RedundantSetter")
    override var isSelectable: Boolean
        get() = false
        set(value) {}

    override fun onRebind() {}

    override var isActivated: Boolean
        get() = itemView.isActivated
        set(value) {
            itemView.isActivated = value
        }
}
